src/rviz/image/mouse_click.cpp

```diff
@@ -0,0 +1,127 @@
+#include "rviz/image/mouse_click.h"
+
+#include <ros/names.h>
+
+
+namespace rviz
+{
+
+MouseClick::MouseClick() : QObject()
+{
+}
+
+MouseClick::~MouseClick()
+{
+}
+
+
+void MouseClick::onInitialize(QObject* parent)
+{
+  setParent(parent);
+  parent->installEventFilter(this);
+
+  has_dimensions_ = false;
+  is_topic_name_ok_ = false;
+  node_handle_.reset(new ros::NodeHandle());
+}
+
+void MouseClick::publish()
+{
+  if (is_topic_name_ok_)
+  {
+    publisher_.reset(
+        new ros::Publisher(node_handle_->advertise<geometry_msgs::PointStamped>(topic_, 1000)));
```

Comment:
Why would you need a queue of 1000 here?

Reply:
That's just a copy paste from other stuff. I will set it to the default.

Reply:
Set queue to 1.